src/main/java/redis/clients/jedis/Observability.java

```diff
@@ -93,8 +93,13 @@ public static void recordTaggedStat(TagKey key, String value, MeasureDouble md,
         }
     }
 
+    // startSpan starts a span and inserts it into the current
+    // scope. It is the caller's responsibility to invoke
+    //      span.end()
+    // when finished using the span.
     public static Span startSpan(String name) {
-        return tracer.spanBuilder(name).startSpan();
+        tracer.spanBuilder(name).startScopedSpan();
+        return tracer.getCurrentSpan();
```

Comment:
@songy23 ditto here

Reply:
Same here, you'll need to keep a reference of scoped span in order to close it.